### PR TITLE
Editorial: Correct the Intl.Locale.prototype.{script,region} algorithms

### LINKED ISSUE #799: Use unicode_script_subtag and unicode_region_subtag in Locale.prototype.{script,region}
The current definition is hard to read and at least for `Intl.Locale.prototype.script` also incorrect.

[get Intl.Locale.prototype.script](https://tc39.es/ecma402/#sec-Intl.Locale.prototype.script)
> 5. Let lang be the longest prefix of locale matched by the unicode_language_id [Unicode locale nonterminal](https://tc39.es/ecma402/#sec-language-tags).
> 6. If lang contains a [subtag](https://tc39.es/ecma402/#bcp-47-subtag) of length 4, return that [subtag](https://tc39.es/ecma402/#bcp-47-subtag).

`new Intl.Locale("de-1996").script` should return the variant subtag `"1996"` according to that definition, which is clearly wrong.

### PR DESCRIPTION
**Pull request by gibson042**
Fixes #799
Ref #735

### CODE REVIEW

File: spec/locale.html [hunk 1]
```diff
@@ -347,8 +347,9 @@ <h1>get Intl.Locale.prototype.region</h1>
         1. Assert: _locale_ can be matched by the <code>unicode_locale_id</code> Unicode locale nonterminal.
         1. Let _lang_ be the longest prefix of _locale_ matched by the <code>unicode_language_id</code> Unicode locale nonterminal.
         1. Let _subtags_ be a List of the subtags of _lang_.
-        1. If the length of _subtags_ is greater than 1 and the length of _subtags_[1] is 4, let _i_ be 2. Otherwise, let _i_ be 1.
-        1. If the length of _subtags_ is greater than _i_ and the length of _subtags_[_i_] is in the inclusive range from 2 to 3, return _subtags_[_i_].
+        1. NOTE: A <code>unicode_region_subtag</code> subtag is only valid immediately after an initial <code>unicode_language_subtag</code> subtag that itself might be followed by a <code>unicode_script_subtag</code> subtag, and <code>unicode_region_subtag</code> cannot be confused with <code>unicode_variant_subtag</code> because their productions are disjoint.
```
Comment (ben-allen) [inline spec/locale.html:350]: This wording feels a little unclear to me. What about:

1. NOTE: A <code>unicode_region_subtag</code> subtag is only valid immediately after an initial <code>unicode_language_subtag</code>, or immediately after a <code>unicode_script_subtag</code> that is itself immediately after an initial <code>unicode_language_subtag</code>

**Review by ben-allen — COMMENTED** [review]
Thank you so much for doing this -- it's a vast improvement to readability

File: spec/locale.html (hunk 1 above)
Comment (gibson042) [reply spec/locale.html]: Acknowledged, and tweaked slightly.

**Review by ben-allen — APPROVED** [review]
LGTM!

**Comment by sffc** [comment]
@gibson042 Is this ready to merge?

**Comment by gibson042** [comment]
I think so, but didn't want to do it unilaterally.

**Pull request merged**